src/sidebar/util/pagination.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * The number of an available pagination page, or `null`, indicating a gap
+ * between sequential numbered pages.
+ *
+ * @typedef {number|null} PageNumber
+ */
+
+/**
+ * Determine the set of (pagination) page numbers that should be provided to
+ * a user, given the current page the user is on, the total number of pages
+ * available, and the number of individual page options desired.
+ *
+ * The first, last and current pages will always be included in the returned
+ * results. Additional pages adjacent to the current page will be added until
+ * `maxPages` is reached. Gaps in the sequence of pages are represented by
+ * `null` values.
+ *
+ * @example
+ *   pageNumberOptions(1, 10, 5) => [1, 2, 3, 4, null, 10]
+ *   pageNumberOptions(3, 10, 5) => [1, 2, 3, 4, null, 10]
+ *   pageNumberOptions(6, 10, 5) => [1, null, 5, 6, 7, null, 10]
+ *   pageNumberOptions(9, 10, 5) => [1, null, 7, 8, 9, 10]
+ *   pageNumberOptions(2, 3, 5) => [1, 2, 3]
+ *
+ * @param {number} totalPages
+ * @param {number} currentPage
+ * @param {number} [maxPages] - The maximum number of numbered pages to make
+ *   available
+ * @return {PageNumber[]} Set of navigation page options to show. `null`
+ *   values represent gaps in the sequence of pages, to be represented later
+ *   as ellipses (...)
+ */
+export function pageNumberOptions(currentPage, totalPages, maxPages = 5) {
+  if (totalPages <= 1) {
+    return [];
+  }
+
+  // Start with first, last and current page. Use a set to avoid dupes.
+  const pageNumbers = new Set([1, currentPage, totalPages]);
+
+  // Fill out the `pageNumbers` with additional pages near the currentPage,
+  // if available
+  let increment = 1;
+  while (pageNumbers.size < Math.min(totalPages, maxPages)) {
+    // Build the set "outward" from the currently-active page
+    if (currentPage + increment <= totalPages) {
+      pageNumbers.add(currentPage + increment);
+    }
+    if (currentPage - increment >= 1) {
+      pageNumbers.add(currentPage - increment);
+    }
+    increment++;
```

Comment:
We have a convention in the code of using prefix increment for everything except loop variables (`++increment`). This makes no difference when the result of the expression is not assigned to anything, it's just a convention.

The rationale is that if the increment is ever used as an expression, you usually want it to evaluate to the new value of the variable rather than the old value.

Reply:
That's quite sensible.